agent/consul/leader_connect_ca_test.go

```diff
@@ -689,6 +689,86 @@ func TestCAManager_Initialize_Vault_WithIntermediateAsPrimaryCA(t *testing.T) {
 	})
 }
 
+func TestCAManager_Verify_NoChangeToSecondaryConfig(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+	ca.SkipIfVaultNotPresent(t)
+
+	vault := ca.NewTestVaultServer(t)
+	vclient := vault.Client()
+	generateExternalRootCA(t, vclient)
+
+	meshRootPath := "pki-root"
+	primaryCert := setupPrimaryCA(t, vclient, meshRootPath)
```

Comment:
This setup is only for a specific scenario (external root CA). Since the bug you are fixing applies to all setups with a secondary DC, I think we can remove this to focus the test on the setup that is required to reproduce the issue.

```suggestion
```